packages/dx-react-scheduler/src/plugins/drag-drop-provider.jsx

```diff
@@ -0,0 +1,329 @@
+
+    const { startTime, endTime } = this.state;
+    if (moment(startTime).isSame(this.appointmentStartTime)
+      && moment(endTime).isSame(this.appointmentEndTime)) return;
+
+    startEditAppointment({ appointmentId: payload.id });
+    changeAppointment({
+      change: {
+        startDate: this.appointmentStartTime,
+        endDate: this.appointmentEndTime,
+      },
+    });
+
+    this.setState({
+      startTime: this.appointmentStartTime,
+      endTime: this.appointmentEndTime,
+      payload,
+    });
+  }
+
+  handleDrop({ commitChangedAppointment, stopEditAppointment }) {
+    const { payload } = this.state;
+    stopEditAppointment({ appointmentId: payload.id });
+    commitChangedAppointment({ appointmentId: payload.id });
+    this.resetCache();
+  }
+
+  render() {
+    const { payload } = this.state;
+    const {
+      containerComponent: Container,
+      draftAppointmentComponent: DraftAppointment,
+      sourceAppointmentComponent: SourceAppointment,
+      allowDrag,
+    } = this.props;
+
+    const draftData = {
+      ...payload, startDate: this.appointmentStartTime, endDate: this.appointmentEndTime,
+    };
+
+    return (
+      <Plugin
+        name="DragDropProvider"
+        dependencies={pluginDependencies}
+      >
+        <Template name="body">
+          <TemplateConnector>
+            {({
+              viewCellsData, startViewDate, endViewDate, excludedDays,
+              timeTableElement, layoutElement, layoutHeaderElement,
+            }, {
+              commitChangedAppointment, changeAppointment,
+              startEditAppointment, stopEditAppointment,
+            }) => (
+              <DragDropProviderCore
+                onChange={this.onPayloadChange({ commitChangedAppointment, stopEditAppointment })}
+              >
+                <DropTarget
+                  onOver={this.onOver({
+                    viewCellsData,
+                    startViewDate,
+                    endViewDate,
+                    excludedDays,
+                    timeTableElement,
+                    layoutElement,
+                    layoutHeaderElement,
+                  }, { changeAppointment, startEditAppointment, stopEditAppointment })}
+                  onDrop={this.onDrop({ commitChangedAppointment, stopEditAppointment })}
+                >
+                  <TemplatePlaceholder />
+                </DropTarget>
+              </DragDropProviderCore>
+            )}
+          </TemplateConnector>
+        </Template>
+
+        <Template
+          name="appointment"
+          predicate={({ data }) => allowDrag(data)}
+        >
+          {params => (
+            <DragSource
+              payload={{ ...params.data, type: params.type }}
+            >
+              {payload && params.data.id === payload.id ? (
+                <SourceAppointment {...params} />
+              ) : (
+                <TemplatePlaceholder params={{ ...params, style: { ...params.style, cursor: 'pointer' } }} />
```

Comment:
Can we get rid of `cursor: 'pointer'` in core plugin?

Reply:
We can pass a `draggingEnabled` property to Appointment Component and use it.

Reply:
I think it's the closest to 'ideology'))

Reply:
I have passed a `clickable` property into an appointment component

Reply:
The component doesn't know about dragging, it know only about itself is clickable.